prog/dftb+/lib_timedep/dynamics.F90

```diff
@@ -3602,11 +3399,582 @@ subroutine getBondPopulAndEnergy(this, bondWork, lastBondPopul, rhoPrim, ham0, o
             & iNeighbour, nNeighbourSK, img2CentCell, iSparseStart)
       end do
       write(fdBondPopul) time * au__fs, sum(bondWork), bondWork
-      if (tWriteAutotest) then
+      if (this%tWriteAutotest) then
         lastBondPopul = sum(bondWork)
       end if
     end if
 
   end subroutine getBondPopulAndEnergy
 
+
+  !> Handles the initializations of the variables needed for the time propagation
+  subroutine initializeDynamics(this, coord, orb, neighbourList, nNeighbourSK,&
+       & iSquare, iSparseStart, img2CentCell, skHamCont, skOverCont, ham, over, env, coordAll,&
+       & H0, spinW, tDualSpinOrbit, xi, thirdOrd, dftbU, onSiteElements,&
+       & refExtPot, solvation, rangeSep, referenceN0, q0, pRepCont, iAtInCentralRegion, &
+       & eigvecsReal, eigvecsCplx, filling, qDepExtPot, tFixEf, Ef, latVec, invLatVec, iCellVec,&
+       & rCellVec, cellVec, speciesAll)
```

Comment:
```suggestion
        & iSquare, iSparseStart, img2CentCell, skHamCont, skOverCont, ham, over, env, coordAll,&
        & H0, spinW, tDualSpinOrbit, xi, thirdOrd, dftbU, onSiteElements,&
        & refExtPot, solvation, rangeSep, referenceN0, q0, pRepCont, iAtInCentralRegion, &
        & eigvecsReal, eigvecsCplx, filling, qDepExtPot, tFixEf, Ef, latVec, invLatVec, iCellVec,&
        & rCellVec, cellVec, speciesAll)

```

Reply:
Thanks a lot for the review @awvwgk ! I addresses all of your comments except for the two changes which will be handled in the next PR to be merged, the non-scc dynamics.